src/plugins/matrixops/slicing_operation.cpp

```diff
@@ -74,27 +74,71 @@ namespace phylanx {namespace execution_tree {    namespace primitives
             docstr
         },
 
+
         match_pattern_type{"tuple_slice",
             std::vector<std::string>{"tuple_slice(_1, _2)"},
             &create_slicing_operation,
             &create_primitive<slicing_operation>,
             docstr
-        }
-      , match_pattern_type{"slice_page",
+        },
+
+        match_pattern_type{"slice_page",
             std::vector<std::string>{"slice_page(_1, __2)"},
             &create_slicing_operation,
             &create_primitive<slicing_operation>,
             docstr
+        },
+
+        match_pattern_type{"slice_row_d",
+            std::vector<std::string>{"slice_row_d(_1, __2)"},
+            &create_slicing_operation,
+            &create_primitive<slicing_operation>,
+            docstr
+        },
+
+        match_pattern_type{"slice_column_d",
+            std::vector<std::string>{"slice_column_d(_1, __2)"},
+            &create_slicing_operation,
+            &create_primitive<slicing_operation>,
+            docstr
+        },
+
+        match_pattern_type{"slice_d",
+            std::vector<std::string>{"slice_d(_1, __2)"},
+            &create_slicing_operation,
+            &create_primitive<slicing_operation>,
+            docstr
         }
+
     };
 
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail
+    {
+        slicing_operation::slice_mode slicing_operation_mode(
+            std::string const& name)
+        {
+            slicing_operation::slice_mode result
+                = slicing_operation::local_mode;
+            if ((name.find("slice_row_d") != std::string::npos) ||
```

Comment:
There is something that I am not sure I understand. Does `slice_d` work on its own?

Reply:
No. I think I need to delete slice_d here.